warehouse/query-core/src/main/java/datawave/query/jexl/visitors/UnmarkedBoundedRangeDetectionVisitor.java

```diff
@@ -22,26 +34,30 @@ public static boolean findUnmarkedBoundedRanges(JexlNode script) {
     }
 
     @Override
-    public Object visit(ASTReference node, Object data) {
-        // determine if we have a marked range that is not actually a range
-        if (QueryPropertyMarker.findInstance(node).isType(BOUNDED_RANGE)) {
-            if (null != data && !JexlASTHelper.findRange().isRange(node)) {
+    public Object visit(ASTAndNode node, Object data) {
+        if (data == null) {
+            return null;
+        }
+
+        // check for a bounded marker where the source node is not a range
+        Instance instance = QueryPropertyMarker.findInstance(node);
+        if (instance.isType(BOUNDED_RANGE)) {
+            LiteralRange<?> range = JexlASTHelper.findRange().getRange(node);
+            if (range == null) {
```

Comment:
this doesn't look quite right. If we have a bounded range and the range is null set hasBounded = true?

Reply:
This is equivalent to the old code. The LiteralRange.isRange() method returns true if the range is not equals to null. The variables are poorly named in this class, if you think a larger refactor would be better for clarity I'm happy to do that as part of this merge request. In essence, this visitor returns true if an unmarked bounded range was found.